Common/src/main/java/com/techarium/techarium/client/screen/MachineCoreScreen.java

```diff
@@ -0,0 +1,241 @@
+				this.minecraft.gameMode.handleInventoryButtonClick(this.menu.containerId, this.buttons.get(i).index);
+				return true;
+			}
+		}
+		if (this.leftPos + 9 <= mouseX && mouseX <= this.leftPos + 23 && this.topPos + 169 <= mouseY && mouseY <= this.topPos + 183) {
+			this.showHints = !this.showHints;
+		}
+		return super.mouseClicked(mouseX, mouseY, mouseButton);
+	}
+
+	@Override
+	public void resize(Minecraft $$0, int $$1, int $$2) {
+		super.resize($$0, $$1, $$2);
+		this.init();
+	}
+
+	@Override
+	public void render(PoseStack poseStack, int mouseX, int mouseY, float partialTicks) {
+		this.renderBackground(poseStack);
+		super.render(poseStack, mouseX, mouseY, partialTicks);
+		this.textInput.render(poseStack, mouseX, mouseY, partialTicks);
+		this.renderTooltip(poseStack, mouseX, mouseY);
+	}
+
+	@Override
+	protected void renderLabels(PoseStack poseStack, int x, int y) {
+		this.font.draw(poseStack, this.title, 6, 3, 16777215);
+	}
+
+	@Override
+	protected void renderBg(PoseStack poseStack, float partialTicks, int mouseX, int mouseY) {
+		RenderSystem.setShaderColor(1.0F, 1.0F, 1.0F, 1.0F);
+		RenderSystem.setShaderTexture(0, TEXTURE);
+		this.blit(poseStack, this.leftPos, this.topPos, 0, 0, this.imageWidth, this.imageHeight);
+		// display scrollbar
+		if (this.buttons.size() > 8) {
+			int startX = this.leftPos + SCROLL_X;
+			int startY = this.topPos + IDS_Y;
+			int scrollAmount = this.buttons.size() - 8;
+			int barSize = 112;
+			int partSize = barSize / (scrollAmount + 1);
+			if (partSize % 2 == 1) {
+				partSize--;
+			}
+			int scrollStartY = startY + partSize * this.startIndex;
+			//blit first line
+			blit(poseStack, startX, scrollStartY, 14, 201, 6, 1);
+			int yStartOffset = 1;
+			// blit the middle line to fill between the start and the end
+			for (int y = scrollStartY + yStartOffset; y < startY + partSize * (this.startIndex + 1) - 3; y += 2) {
+				blit(poseStack, startX, y, 14, 202, 6, 2);
+			}
+			// blit the 2 end lines
+			blit(poseStack, startX, scrollStartY + partSize - 3, 14, 207, 6, 3);
+		}
+		// display hint button
+		if (this.showHints) {
+			this.blit(poseStack, this.leftPos + 9, this.topPos + 169, 0, 196, 14, 14);
+		}
+		this.font.draw(poseStack, SHOW_HINTS_TEXT, this.leftPos + 31, this.topPos + 172, 4210752);
+	}
+
+	private static class MultiBlockButton extends Button implements Comparable<MultiBlockButton> {
+
+		/**
+		 * The index in the multiblock registry.
+		 */
+		private final int index;
+		/**
+		 * If the button is selected.
+		 */
+		private boolean selected;
+
+		public MultiBlockButton(int index, int x, int y, Component name, OnPress onPress) {
+			super(x, y, 100, 14, name, onPress);
+			this.index = index;
+		}
+
+		@Override
+		public void renderButton(PoseStack poseStack, int mouseX, int mouseY, float partialTicks) {
+			RenderSystem.setShaderColor(1.0F, 1.0F, 1.0F, 1.0F);
```

Comment:
This should set the texture shader as other widgets might not have texture shader set already.